src/utils.cpp

```diff
@@ -64,9 +64,9 @@ bool greet()
     std::cout << "Hello, Welcome to our program" << std::endl;
```

Comment:
Maybe keep usage of `std::endl` vs `\n` consistent? The recommendation is to use `\n` because `std::endl` always flushes the buffer.